runtime/parachains/src/runtime_api_impl/v1.rs

```diff
@@ -231,6 +232,30 @@ pub fn session_index_for_child<T: initializer::Config>() -> SessionIndex {
 	<shared::Module<T>>::session_index()
 }
 
+/// Implementation for the `AuthorityDiscoveryApi::authorities()` function of the runtime API.
+/// It is a heavy call, but currently only used for authority discovery, so it is fine.
+/// Gets current and historical authority ids using session_info module.
+pub fn get_historical_and_current_authority_ids<T: initializer::Config + pallet_authority_discovery::Config>() -> Vec<AuthorityDiscoveryId> {
+	let current_session_index = session_index_for_child::<T>();
+	let earliest_stored_session = <session_info::Module<T>>::earliest_stored_session();
+	let mut authority_ids = vec![];
+
+	for session_index in earliest_stored_session..=current_session_index {
+		let info = <session_info::Module<T>>::session_info(session_index);
+		if let Some(mut info) = info {
+			authority_ids.append(&mut info.discovery_keys);
+		}
+	}
+
+	let mut next_authorities = <pallet_authority_discovery::Module<T>>::next_authorities();
```

Comment:
session_info module defines `AuthorityDiscoveryConfig` trait to help with mock testing,
we can add `next_authorities` method to it, this would avoid having the `pallet_authority_discovery::Config` bound here

and call it
```rust
<T as session_info::AuthorityDiscoveryConfig>::next_authorities()
```
also this can go into initialization, i.e. instead of
```rust
let mut authority_ids = vec![];
```

Reply:
@ordian IMHO I think it would be clearer to have explicit trait bound rather than use the trait impl that is primarily used for testing, as it will create shared dependency between mock testing and actual code.

Reply:
I don't agree it's primarily used for tests, rather it is to allow testing as implementing `pallet_authority_discovery::Config` manually is much harder. But it's also not that important.